components/webac/src/main/java/org/trellisldp/webac/WebAcService.java

```diff
@@ -79,10 +84,17 @@
     public static final String CONFIG_WEBAC_MEMBERSHIP_CHECK = "trellis.webac.membership.check";
 
     private static final Logger LOGGER = getLogger(WebAcService.class);
+    private static final CompletionStage<Void> DONE = CompletableFuture.completedFuture(null);
     private static final RDF rdf = getInstance();
     private static final IRI root = rdf.createIRI(TRELLIS_DATA_PREFIX);
+    private static final IRI rootAuth = rdf.createIRI(TRELLIS_DATA_PREFIX + "#auth");
     private static final Set<IRI> allModes = new HashSet<>();
 
+    /** The permissive Authorizations in effect when no ACL is present on the root node. **/
+    private static final List<Authorization> defaultRootAuthorizations =
+        unmodifiableList(singletonList(Authorization.from(rootAuth, generateDefaultRootAuthorizations()
```

Comment:
Minor: the `generateDefaultRootAuthorizations` method call here generates a `Dataset` but never closes it. Perhaps it would be better to create an additional `static` method such as:

```java
private static final List<Authorization> getAuthorizations() {
    try (final Dataset dataset = generateDefaultRootAuthorizations()) {
        dataset.getGraph(Trellis.PreferAccessControl).map(graph -> Authorization.from(rootAuth, graph)
                .map(Collections::singletonList).orElseGet(emptyList());
    } catch (final Exception ex) {
        throw new RuntimeTrellisException("...", ex);
    }
}
```

And then you can just call:

```java
private static final List<Authorization> defaultRootAuthorizations = unmodifiableList(getAuthorizations());
```

Feel free to change names, etc.

Reply:
I am reusing the Dataset for the root node ACL persistence and the "fall back ACL". So I didn't want to close it, because I keep it around. If you close it I believe you can no longer call getGraph(), etc...

Reply:
Aha, I think you are suggesting that I keep the authorizations and not the Dataset, which works for the fall back case just fine. Let me try that.

Reply:
This is (really!) a good motivation for us to tackle https://github.com/trellis-ldp/trellis/issues/358 and https://github.com/orgs/trellis-ldp/projects/4. The truth is that `Dataset::close` is totally unclear. Its comments include ` The behaviour of the other dataset methods are undefined after closing the dataset.` Having said that, I've done nothing to act on this stuff, so I'll shut up now and go back and keep working on `trellis-cassandra`.